src/types/validate.go

```diff
@@ -0,0 +1,334 @@
+		if component.DeprecatedGroup != "" {
+			if component.Default {
+				if _, ok := groupDefault[component.DeprecatedGroup]; ok {
+					return fmt.Errorf(lang.PkgValidateErrGroupMultipleDefaults, component.DeprecatedGroup, groupDefault[component.DeprecatedGroup], component.Name)
+				}
+				groupDefault[component.DeprecatedGroup] = component.Name
+			}
+			groupedComponents[component.DeprecatedGroup] = append(groupedComponents[component.DeprecatedGroup], component.Name)
+		}
+	}
+
+	for groupKey, componentNames := range groupedComponents {
+		if len(componentNames) == 1 {
+			return fmt.Errorf(lang.PkgValidateErrGroupOneComponent, groupKey, componentNames[0])
+		}
+	}
+
+	return nil
+}
+
+// Validate runs all validation checks on component actions.
+func (a ZarfComponentActions) Validate() error {
+	if err := a.OnCreate.Validate(); err != nil {
+		return fmt.Errorf(lang.PkgValidateErrAction, err)
+	}
+
+	if a.OnCreate.HasSetVariables() {
+		return fmt.Errorf("cannot contain setVariables outside of onDeploy in actions")
+	}
+
+	if err := a.OnDeploy.Validate(); err != nil {
+		return fmt.Errorf(lang.PkgValidateErrAction, err)
+	}
+
+	if a.OnRemove.HasSetVariables() {
+		return fmt.Errorf("cannot contain setVariables outside of onDeploy in actions")
+	}
+
+	return nil
+}
+
+// ValidateImportDefinition validates the component trying to be imported.
+func (c ZarfComponent) ValidateImportDefinition() error {
+	path := c.Import.Path
+	url := c.Import.URL
+
+	// ensure path or url is provided
+	if path == "" && url == "" {
+		return fmt.Errorf(lang.PkgValidateErrImportDefinition, c.Name, "neither a path nor a URL was provided")
+	}
+
+	// ensure path and url are not both provided
+	if path != "" && url != "" {
+		return fmt.Errorf(lang.PkgValidateErrImportDefinition, c.Name, "both a path and a URL were provided")
+	}
+
+	// validation for path
+	if url == "" && path != "" {
+		// ensure path is not an absolute path
+		if filepath.IsAbs(path) {
+			return fmt.Errorf(lang.PkgValidateErrImportDefinition, c.Name, "path cannot be an absolute path")
+		}
+	}
+
+	// validation for url
+	if url != "" && path == "" {
+		ok := helpers.IsOCIURL(url)
+		if !ok {
+			return fmt.Errorf(lang.PkgValidateErrImportDefinition, c.Name, "URL is not a valid OCI URL")
+		}
+	}
+
+	return nil
+}
+
+// HasSetVariables returns true if any of the actions contain setVariables.
+func (as ZarfComponentActionSet) HasSetVariables() bool {
+	check := func(actions []ZarfComponentAction) bool {
+		for _, action := range actions {
+			if len(action.SetVariables) > 0 {
+				return true
+			}
+		}
+		return false
+	}
+
+	return check(as.Before) || check(as.After) || check(as.OnSuccess) || check(as.OnFailure)
+}
+
+// Validate runs all validation checks on component action sets.
+func (as ZarfComponentActionSet) Validate() error {
+	validate := func(actions []ZarfComponentAction) error {
+		for _, action := range actions {
+			return action.Validate()
```

Comment:
this should be `if err := action.Validate(); err != nil`, otherwise this function exits before it has validated all `actions`